ramls/event.json

```
@@ -0,0 +1,55 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Event data model",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "id": {
+      "description": "UUID",
+      "type": "string"
+    },
+    "eventType": {
+      "description": "Unique Event type, human-readable String, not UUID",
+      "type": "string"
+    },
+    "metadata": {
+      "description": "Event metadata",
+      "type": "object",
+      "properties": {
+        "tenantId" : {
+          "description": "Tenant id",
+          "type": "string"
+        },
+        "eventTTL": {
+          "description": "Time-to-live (TTL) for event in minutes",
+          "type": "integer"
+        },
+        "correlationId": {
+          "description": "Id to track related events",
+          "type": "string"
+        },
+        "originalEventId": {
+          "description": "Id of the event that started the sequence of related events",
+          "type": "string"
+        },
+        "publisherCallback": {
```

What is the expected contents of this property? Is it the choice between an endpoint or error event type?

If it is the choice, where is the endpoint itself or the event type specified?

If it is either an endpoint or an event type, how will the system differentiate between them?

---

The original idea was to parse the content and differentiate the url or string, but you're right, we could probably make it more clear, something like the following:         
```suggestion
          "publisherCallback": {
          "description": "Allows a publisher to provide a callback endpoint or an error Event Type to be notified that despite the fact that there are subscribers for such an event type no one has received the event within the specified period of time",
          "type": "object",
          "properties": {
            "endpoint": {
              "description": "Callback endpoint",
              "type": "string"
            },
            "eventType": {
              "description": "Error Event Type",
              "type": "string"
            }
          }
        }
```

---

I personally think an explicit distinction is much better 🙂